### PR TITLE
Refactor Cloud connection refactor data source and documentation

### PR DESCRIPTION
**Pull request by Diptipowervs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: website/docs/d/pi_cloud_connection.html.markdown [hunk 1]
```diff
@@ -37,28 +32,26 @@ data "ibm_pi_cloud_connection" "example" {
...
 - `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
 - `pi_cloud_connection_name` - (Required, String) The cloud connection name to be used.
 
 ## Attribute reference
-
 In addition to all argument reference list, you can access the following attribute references after your data source is created.
 
-- `id` - (String) The unique identifier of the cloud connection.
-- `classic_enabled` - (Bool) Is classic endpoint destination enabled?
+- `classic_enabled` - (Bool) Enable classic endpoint destination.
```
Comment (ismirlia) [inline website/docs/d/pi_cloud_connection.html.markdown:43]: Change to (Boolean) to match other parts of the IBM terraform provider documentation.

File: website/docs/d/pi_cloud_connection.html.markdown [hunk 2]
```diff
@@ -37,28 +32,26 @@ data "ibm_pi_cloud_connection" "example" {
...
+- `metered` - (String) Enable metering for this cloud connection.
+- `networks` - (Set) Set of Networks attached to this cloud connection.
 - `port` - (String) Port.
 - `speed` - (Integer) Speed of the cloud connection (speed in megabits per second).
 - `status` - (String) Link status.
 - `user_ip_address` - (String) User IP address.
-- `vpc_crns` - (Set of String) Set of VPCs attached to this cloud connection.
-- `vpc_enabled` - (Bool) Is VPC enabled for this cloud connection?
+- `vpc_crns` - (Set) Set of VPCs attached to this cloud connection.
+- `vpc_enabled` - (Bool) Enable VPC for this cloud connection.
```
Comment (ismirlia) [inline website/docs/d/pi_cloud_connection.html.markdown:57]: Change to (Boolean) to match other parts of the IBM terraform provider documentation.

File: ibm/service/power/data_source_ibm_pi_cloud_connection.go [hunk 3]
```diff
@@ -128,8 +116,8 @@ func dataSourceIBMPICloudConnectionRead(ctx context.Context, d *schema.ResourceD
 		return diag.FromErr(err)
 	}
 
-	cloudInstanceID := d.Get(helpers.PICloudInstanceId).(string)
-	cloudConnectionName := d.Get(helpers.PICloudConnectionName).(string)
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	cloudConnectionName := d.Get(Attr_Name).(string)
```
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_cloud_connection.go:120]:
```suggestion
	cloudConnectionName := d.Get(Arg_CloudConnectionName).(string)
```

**Review by yussufsh — CHANGES_REQUESTED** [review]

**Comment by yussufsh** [comment]
@Diptipowervs please run and paste the test results here.

File: ibm/service/power/data_source_ibm_pi_cloud_connection.go (hunk 3 above)
Comment (Diptipowervs) [reply ibm/service/power/data_source_ibm_pi_cloud_connection.go:120]: Changes made

**Comment by Diptipowervs** [comment]
> @Diptipowervs please run and paste the test results here.
After making commit.The output is as below

--- PASS: TestAccIBMPICloudConnectionDataSource_basic (153.76s)
PASS

File: website/docs/d/pi_cloud_connections.html.markdown [hunk 4]
```diff
@@ -36,32 +32,29 @@ data "ibm_pi_cloud_connections" "example" {
...
 ## Attribute reference
-
 In addition to all argument reference list, you can access the following attribute references after your data source is created.
 
 - `connections` - (List) List of all the Cloud Connections.
 
   Nested scheme for `connections`:
-
-  - `classic_enabled` - (Bool) Is Classic endpoint destination enabled.
+  - `classic_enabled` - (Bool) Enable classic endpoint destination.
```
Comment (yussufsh) [inline website/docs/d/pi_cloud_connections.html.markdown:45]:
```suggestion
  - `classic_enabled` - (Boolean) Enable classic endpoint destination.
```

File: website/docs/d/pi_cloud_connections.html.markdown [hunk 5]
```diff
@@ -36,32 +32,29 @@ data "ibm_pi_cloud_connections" "example" {
...
+  - `networks` - (Set) Set of Networks attached to this cloud connection.
   - `port` - (String) Port.
   - `speed` - (Integer) Speed of the cloud connection (speed in megabits per second).
   - `status` - (String) Link status.
   - `user_ip_address` - (String) User IP address.
-  - `vpc_crns` - (Set of String) Set of VPCs attached to this cloud connection.
-  - `vpc_enabled` - (Bool) Is VPC enabled for this cloud connection.
-  - `connection_mode` - (String) Type of service the gateway is attached to.
+  - `vpc_crns` - (Set) Set of VPCs attached to this cloud connection.
+  - `vpc_enabled` - (Bool) Enable VPC for this cloud connection.
```
Comment (yussufsh) [inline website/docs/d/pi_cloud_connections.html.markdown:60]:
```suggestion
  - `vpc_enabled` - (Boolean) Enable VPC for this cloud connection.
```

File: website/docs/d/pi_cloud_connection.html.markdown [hunk 6]
```diff
@@ -20,15 +18,12 @@ data "ibm_pi_cloud_connection" "example" {
...
-
 - Please find [supported Regions](https://cloud.ibm.com/apidocs/power-cloud#endpoint) for endpoints.
 - If a Power cloud instance is provisioned at `lon04`, The provider level attributes should be as follows:
-
   - `region` - `lon`
   - `zone` - `lon04`
 
-  Example usage:
-
+Example usage:
```
Comment (yussufsh) [inline website/docs/d/pi_cloud_connection.html.markdown:26]: This example is part of the second bullet point—no need to remove a tab.

File: ibm/service/power/data_source_ibm_pi_cloud_connection_test.go [hunk 7]
```diff
@@ -30,9 +30,8 @@ func TestAccIBMPICloudConnectionDataSource_basic(t *testing.T) {
...
 	return fmt.Sprintf(`
-	data "ibm_pi_cloud_connection" "example" {
-		pi_cloud_connection_name 	= "%s"
-		pi_cloud_instance_id 		= "%s"
-	}
-	`, acc.PiCloudConnectionName, acc.Pi_cloud_instance_id)
+		data "ibm_pi_cloud_connection" "example" {
+			pi_cloud_connection_name 	= "%s"
+			pi_cloud_instance_id 		= "%s"
+		}`, acc.PiCloudConnectionName, acc.Pi_cloud_instance_id)
```
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_cloud_connection_test.go:36]: Why a new line?

**Review by yussufsh — APPROVED** [review]

**Comment by yussufsh** [comment]
@ismirlia lgtm, but would like your review as well.

File: ibm/service/power/data_source_ibm_pi_cloud_connection_test.go (hunk 7 above)
Comment (ismirlia) [reply ibm/service/power/data_source_ibm_pi_cloud_connection_test.go:36]: This is not a new line, this part of the same line. The line number is the same, but view is squished so it appears to be on a new line.

File: website/docs/d/pi_cloud_connection.html.markdown (hunk 6 above)
Comment (ismirlia) [reply website/docs/d/pi_cloud_connection.html.markdown:26]: It's the same problem that was in the list-catalog refactor. This is a common problem for 90% of the existing markdown files.

The change changes it from:
Screenshot 2024-01-24 at 9 08 16 AM

to:
Screenshot 2024-01-24 at 9 09 01 AM

**Review by hkantare — APPROVED** [review]

**Comment by Diptipowervs** [comment]
--- PASS: TestAccIBMPICloudConnectionDataSource_basic (30.26s)
PASS

**Pull request merged**